src/ServiceInsight/Framework/WorkNotifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace ServiceInsight.Framework
+{
+    using System;
+    using System.Reactive.Disposables;
+    using Caliburn.Micro;
+    using ServiceInsight.Framework.Events;
+
+    public interface IWorkNotifier
+    {
+        IDisposable NotifyOfWork();
+
+        IDisposable NotifyOfWork(string startMessage);
+
+        IDisposable NotifyOfWork(string startMessage, string finishMessage);
+    }
+
+    class WorkNotifier : IWorkNotifier
+    {
+        IEventAggregator eventAggregator;
+
```

Comment:
The WorkTracker interface tells you one thing that WorkNotifier is now managing: "Is there any work in progress"? To me it feels if WorkNotifier encapsulated its state, it'd know if there are work in progress or not and could answer that question. 

The subscriber would be the Shell where it just updates the UI with the messages where as now it is also answering if there are currently work in progress or not.